aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -328,6 +334,27 @@ public SdkHttpFullRequest sign(
     return mutableRequest.build();
   }
 
+  private String body(SdkHttpFullRequest request) {
+    if (shouldAddBody(request) && request.contentStreamProvider().isPresent()) {
+      try (InputStream is = request.contentStreamProvider().get().newStream()) {
+        return IoUtils.toUtf8String(is);
+      } catch (IOException e) {
+        throw new UncheckedIOException("Failed to set body for S3 sign request", e);
```

Comment:
I think we do want to fail. The body is optional from an S3 signing spec perspective, but it's required for DeleteObjectsRequest. This block will currently be only reached for DeleteObjectsRequest, so we get the `IOException` I think we want to fail fast to a user rather than get rejected on the server side. Lmk your thoughts, cc: @danielcweeks

Reply:
I think it's possible that you could sign if the body isn't present (it really depends on the level of checking the service performs).  I feel like we would be ok logging and letting the service deny if the body isn't present.  I would probably put it at a much lower logging level (debug?) because it will either be ok without the body, or be rejected.

Reply:
In practice, the body will probably always be there.  I doubt the SDK has a lot of options for modifying this behavior.

Reply:
Good point, yeah ultimately it's up to the server implementation for signing. Will update to have a debug log in case there's an `IOException` thrown.

Reply:
logging at debug makes sense to me